gems/aws-sigv4/spec/signer_spec.rb

```diff
@@ -143,6 +143,38 @@ module Sigv4
           expect(signature.headers['host']).to eq('domain.com:123')
         end
 
+        it 'includes WS port in Host when not 80' do
+          signature = Signer.new(options).sign_request(
+            http_method: 'GET',
+            url: 'ws://domain.com:123'
+          )
+          expect(signature.headers['host']).to eq('domain.com:123')
+        end
+
+        it 'includes WSS port in Host when not 443' do
+          signature = Signer.new(options).sign_request(
+            http_method: 'GET',
+            url: 'wss://domain.com:123'
+          )
+          expect(signature.headers['host']).to eq('domain.com:123')
+        end
+
+        it 'does not include trailing colon in HOST if WS default port' do
```

Comment:
I like that you made these tests about trailing colon, but I would see if you can refactor/delete these tests with the new changes. The previous tests were for HTTP, and now WebSocket, but we can have cases for: "scheme default port and uri port are the same", "scheme default port and uri port are different", "scheme not recognized and uri port not provided", and "scheme not recognized and uri port provided". You can rename those tests (or even use an RSpec 'context' block for 'scheme not recognized by URI module') if you'd like.